handlers/secrets.go

```diff
@@ -1,28 +1,148 @@
 package handlers
 
 import (
+	"encoding/json"
 	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
 	"net/http"
 
 	"github.com/openfaas/faas/gateway/requests"
 	apiv1 "k8s.io/api/core/v1"
 	v1beta1 "k8s.io/api/extensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 )
 
 const (
 	secretsMountPath = "/var/openfaas/secrets"
+	secretLabel      = "app.kubernetes.io/managed-by"
+	secretLabelValue = "openfaas"
 )
 
 // MakeSecretHandler makes a handler for Create/List/Delete/Update of
 //secrets in the Kubernetes API
-func MakeSecretHandler() http.HandlerFunc {
+func MakeSecretHandler(namespace string, kube kubernetes.Interface) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		defer r.Body.Close()
-		w.WriteHeader(http.StatusNotImplemented)
-		w.Write([]byte("Not implemented\n"))
+		if r.Body != nil {
+			defer r.Body.Close()
+		}
+
+		switch r.Method {
+		case http.MethodGet:
+			selector := fmt.Sprintf("%s=%s", secretLabel, secretLabelValue)
```

Comment:
Please could you create a new method to prevent this handler making maintenance harder?

i.e. `func listSecrets(w http.Response, r *http.Request)` would be fine.

Reply:
@alexellis  please take a look, I split it out. let me know if i should restructure it a little more.